### PR TITLE
Cairo1 events

### PR DESCRIPTION
**Pull request by notV4l**
Add cairo1 events test (https://github.com/0xSpaceShard/starknet-hardhat-plugin/issues/378). Linked with https://github.com/0xSpaceShard/starknet-hardhat-plugin/pull/380

- add a `test/cairo1` directory
- moved `declare-v2-test` to test/cairo1 directory
- add starknet.js ^4.22.0 to handle uint256

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,6 +28,7 @@
   },
   "dependencies": {
     "axios": "^1.0.0",
-    "hardhat": "2.14.0"
+    "hardhat": "2.14.0",
+    "starknet": "^4.22.0"
```
Comment (FabijanC) [inline package.json:32]: Is this a necessary change? I feel like it isn't.

File: cairo1-contracts/events.cairo [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    fn ComplexEvent(simple: felt252, event_struct: EventStruct, type_tuple: (felt252, u8, usize)) {}
+
+    #[constructor]
+    fn constructor(initial_balance: felt252) {
+        balance::write(initial_balance);
+    }
+
+    // to force EventStruct type generation in abi
+    #[external]
+    fn useless(event_struct: EventStruct) {}
```
Comment (FabijanC) [inline cairo1-contracts/events.cairo:36]: If this is going to be the recommended way, it needs to be documented. Has Starkware been informed about this? Is this fixed in cairo 2.0.0 or a newer version? Are event specs perhaps stored in some other part of the JSON compilation artifact, and not in the ABI?

File: cairo1-contracts/events.cairo [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        let event_struct = EventStruct {
+            type_felt252: 'abc',
+            type_u8: 1_u8,
+            type_u16: 2_u16,
+            type_u32: 3_u32,
+            type_u64: 4_u64,
+            type_u128: 5_u128,
+            type_u256: u256 {
+                low: 0, high: 1
+            }, type_tuple: ('tuple', 1_u8),
```
Comment (FabijanC) [inline cairo1-contracts/events.cairo:58]: Did your formatter make it look like this? Can the `type_tuple` be in a separate line?

File: test/cairo1/decode-events.test.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+                    simple: starknet.shortStringToBigInt("simple"),
+                    event_struct: {
+                        type_felt252: starknet.shortStringToBigInt("abc"),
+                        type_u8: 1n,
+                        type_u16: 2n,
+                        type_u32: 3n,
+                        type_u64: 4n,
+                        type_u128: 5n,
+                        type_u256: BigInt(uint256.uint256ToBN({ low: 0n, high: 1n }).toString()),
+                        // type_array_u8: [1n, 2n, 3n],
```
Comment (FabijanC) [inline test/cairo1/decode-events.test.ts:59]: What's with this comment?

**Review by FabijanC — CHANGES_REQUESTED** [review]
I like the way you wrote the tests. Let's discuss a few of points in the comments which I left.

File: test/cairo1/decode-events.test.ts (hunk 4 above)
Comment (notV4l) [reply test/cairo1/decode-events.test.ts:59]: it was attempt to test arrays but it fails at compilation

File: cairo1-contracts/events.cairo (hunk 2 above)
Comment (notV4l) [reply cairo1-contracts/events.cairo:36]: i created an issue about this : https://github.com/starkware-libs/cairo/issues/3419
from my quick tests it was still an issue in 2.0.0-rc2

Comment (FabijanC) [reply cairo1-contracts/events.cairo:36]: Right, I forgot about the issue. Can you put the link as a comment between L34 and L35? And I'll ping Starkware about it.

Comment (notV4l) [reply cairo1-contracts/events.cairo:37]: done

File: package.json (hunk 1 above)
Comment (notV4l) [reply package.json:32]: added

**Pull request merged**